### PR TITLE
Fast sync - deprecation warning 

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
Deprecation messages for FAST sync mode per https://www.lfdecentralizedtrust.org/blog/sunsetting-tessera-and-simplifying-hyperledger-besu

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java [hunk 1]
```diff
@@ -52,27 +52,34 @@
...
 public class MiningOptions implements CLIOptions<MiningConfiguration> {
 
+  private static final String DEPRECATION_PREFIX =
+      "Deprecated. PoW consensus is deprecated. See CHANGELOG for alternative options. ";
+
   @Option(
       names = {"--miner-enabled"},
-      description = "Set if node will perform mining (default: ${DEFAULT-VALUE})")
+      description =
+          DEPRECATION_PREFIX + "Set if node will perform mining (default: ${DEFAULT-VALUE})")
```
Comment (fab-10) [inline besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:61]: This is not related to fast sync, should be another PR?

**Review by fab-10 — COMMENTED** [review]
Other than fast sync, there are deprecation warnings for miner options and methods, should they go to another PR?

**Comment by macfarla** [comment]
> ~Other than fast sync, there are deprecation warnings for miner options and methods, should they go to another PR?~
> 
> I see that this is built on top of #7905, fine to approve after that is merged

Thanks @fab-10 all up to date now just needs approval
